src/main/java/org/opentripplanner/middleware/models/OtpUser.java

```diff
@@ -113,6 +129,28 @@ public boolean delete(boolean deleteAuth0User) {
             }
         }
 
+        // If a related user, invalidate relationship with all dependents.
+        for (String userId : dependents) {
+            OtpUser dependent = Persistence.otpUsers.getById(userId);
+            if (dependent != null) {
+                for (RelatedUser relatedUser : dependent.relatedUsers) {
+                    if (relatedUser.email.equals(this.email)) {
+                        relatedUser.status = RelatedUser.RelatedUserStatus.INVALID;
+                    }
+                }
+                Persistence.otpUsers.replace(dependent.id, dependent);
+            }
+        }
+
+        // If a dependent, remove relationship with all related users.
+        for (RelatedUser relatedUser : relatedUsers) {
```

Comment:
As indicated, these changes should probably also take place when a users submits a new list of related users where some related users have been removed.

Reply:
Addressed in: https://github.com/ibi-group/otp-middleware/pull/256/commits/4a00bfa58c2f747e1f645b4b2b0a4739ac136b6f